File: day2/index1.js
const fs = require("fs");

const matrixStr = fs.readFileSync("./input", "utf-8").trim();

const matrix = matrixStr
  .split("\n")
  .map(str => str.split("\t").map(n => parseInt(n, 10)));

const sum = a => a.reduce((x, y) => x + y, 0);

const cs = sum(matrix.map(r => Math.max(...r) - Math.min(...r)));

console.log(cs);

File: clojure/day02/index1.clj
(require '[clojure.string :as s])

(def parse-int #(Integer/parseInt %))

(defn split [p s] (s/split s p))

(def matrix-str (slurp "./input"))

(defn parse-row [s] (->> s
  (split #"\t")
  (map parse-int)
))

(def matrix (->> matrix-str
  s/split-lines
  (map parse-row)
))

(defn sum [a] (reduce + 0 a))

(println (->> matrix
  (map (fn [r] (- (apply max r) (apply min r))))
  sum
))
